### PR TITLE
fix: use correct key length when looking up foreign keys

### PR DESCRIPTION
**Pull request by lenaschoenburg**
Previously we used the full key buffer when looking up foreign keys. This was wrong, as the capacity is not related to the actual key length.

Instead we now calculate the correct key length based on the column family prefix and the actual inner key length.

@pihme Btw, you had the right idea that the other tests were not strict enough: https://github.com/camunda-cloud/zeebe/pull/8906#discussion_r826200697

### CODE REVIEW

File: zb-db/src/test/java/io/camunda/zeebe/db/impl/rocksdb/transaction/ForeignKeyCheckerTest.java [hunk 1]
```diff
@@ -58,6 +62,33 @@ void shouldSucceedOnExistingForeignKey() throws Exception {
...
+        new ForeignKeyChecker(
+            (ZeebeTransactionDb<?>) db, new ConsistencyChecksSettings(true, true));
+
+    // when -- key 1 exists in first column family
+    final var cf1Key = new DbLong();
+    cf1Key.wrapLong(1);
+    cf1.insert(cf1Key, DbNil.INSTANCE);
+
+    // then -- referring to key 1 does not throw
+    check.assertExists(
```
Comment (pihme) [inline zb-db/src/test/java/io/camunda/zeebe/db/impl/rocksdb/transaction/ForeignKeyCheckerTest.java:85]: :wrench: there is something like `Assertions.assertNoExceptionThrownBy( lambda)`

**Review by pihme — APPROVED** [review]
nice catch

One suggestion for the test

File: zb-db/src/test/java/io/camunda/zeebe/db/impl/rocksdb/transaction/ForeignKeyCheckerTest.java (hunk 1 above)
Comment (oleschoenburg) [reply zb-db/src/test/java/io/camunda/zeebe/db/impl/rocksdb/transaction/ForeignKeyCheckerTest.java:85]: Oh my, I've been looking for this forever but somehow couldn't find. Thanks :+1:

Comment (oleschoenburg) [reply zb-db/src/test/java/io/camunda/zeebe/db/impl/rocksdb/transaction/ForeignKeyCheckerTest.java:85]: It's `assertDoesNotThrow` btw.

**Pull request merged**